doc/source/train/user_guide.rst

```diff
@@ -652,28 +716,127 @@ Checkpoints can be loaded into the training function in 2 steps:
+        from torch.optim import Adam
+        import numpy as np
+
+
+        def train_func(config):
+            n = 100
+            # create a toy dataset
+            # data   : X - dim = (n, 4)
+            # target : Y - dim = (n, 1)
+            Y = torch.Tensor(np.random.uniform(0, 1, size=(n, 1)))
+            X = torch.Tensor(np.random.normal(0, 1, size=(n, 4)))
+
+            # toy neural network : 1-layer
+            model = nn.Linear(4, 1)
+            mse = nn.MSELoss()
+            optimizer = Adam(model.parameters(), lr=3e-4)
+            start_epoch = 0
+
+            checkpoint = train.load_checkpoint()
+            if checkpoint:
+                # assume that we have run the train.save_checkpoint() example
+                #                and successfully save some model weights
+                model.load_state_dict(checkpoint.get("model_weights"))
+                start_epoch = checkpoint.get("epoch", -1) + 1
+
+            # wrap the model in DDP
+            model = ray.train.torch.prepare_model(model)
+            for epoch in range(start_epoch, config["num_epochs"]):
+                y = model.forward(X)
+                # compute loss
+                loss = mse(y, Y)
+                # back-propagate loss
+                optimizer.zero_grad()
+                loss.backward()
+                optimizer.step()
+                ddp_state_dict = model.state_dict()
+                consume_prefix_in_state_dict_if_present(ddp_state_dict, "module.")
+                train.save_checkpoint(epoch=epoch, model_weights=ddp_state_dict)
+
+
+        trainer = Trainer(backend="torch", num_workers=2)
+        trainer.start()
+        # save a checkpoint
+        trainer.run(train_func, config={"num_epochs": 2})
+        # load a checkpoint
+        trainer.run(train_func, config={"num_epochs": 4},
+                    checkpoint=trainer.latest_checkpoint)
+
+        trainer.shutdown()
+
+        print(trainer.latest_checkpoint)
 
-    print(trainer.latest_checkpoint)
-    # {'epoch': 4, 'model': 10}
+
+  .. group-tab:: TensorFlow
+
+    .. code-block:: python
+
+        from ray import train
+        from ray.train import Trainer
+
+        import numpy as np
+
+
+        def train_func(config):
+            import tensorflow as tf
+            n = 100
+            # create a toy dataset
+            # data   : X - dim = (n, 4)
+            # target : Y - dim = (n, 1)
+            Y = np.random.uniform(0, 1, size=(n, 1))
+            X = np.random.normal(0, 1, size=(n, 4))
+
+            start_epoch = 0
+            strategy = tf.distribute.experimental.MultiWorkerMirroredStrategy()
+
+            with strategy.scope():
+                # toy neural network : 1-layer
+                model = tf.keras.Sequential([tf.keras.layers.Dense(1, activation="linear")])
+                checkpoint = train.load_checkpoint()
+                if checkpoint:
+                    model.build((None, 4))
```

Comment:
You can get rid of the `build` step by adding `input_shape` - I think you need this for the non-checkpoint flow as well!

```suggestion
                model = tf.keras.Sequential([tf.keras.layers.Dense(1, activation="linear", input_shape=(4,))])
                checkpoint = train.load_checkpoint()
                if checkpoint:
```

Reply:
Hi @matthewdeng, thank you for your suggestion! It looks better to add input shape. I will update the example shortly.

For non-checkpoint flow, the input size will be inferred when we call `fit()`, `eval()` and `predict()` for the first time (see the last example in [Examples](https://www.tensorflow.org/api_docs/python/tf/keras/Sequential#examples_3)).